lib/puppet/provider/registry_value/registry.rb

```diff
@@ -174,6 +174,65 @@ def write_value
     end
   end
 
+  BITS_PER_BYTE = 8
+  BYTE_MASK = 0xFF
+
+  def to_byte_array(num, of_length)
+    bytes = []
+    while
+      # mask only least significant byte and prepend
+      bytes.unshift(num & BYTE_MASK)
+
+      # nothing left to shave off
+      break if (num <= BYTE_MASK)
+
+      # shift off the least significant byte and continue
+      num >>= BITS_PER_BYTE
+    end
+
+    pad = of_length - bytes.length
+    bytes.concat([0] * pad) if pad > 0
+
+    bytes
+  end
```

Comment:
Yeah, I had done a quick pack / unpack approach, but thought that my conversion code would be easier to understand / simpler to maintain (given we need to convert numeric to string then to bytes).

However, when writing up what I had done as a note here, I realize that I made a critical error.

Addressing now, and re-releasing module.


Thanks for making me double check this.